### PR TITLE
feat(policies): add rails cookie leaks policy

### PR DESCRIPTION
**Pull request by elsapet**
## Description
Add cookie leaking policy for Rails

## Checklist

- [x] I've added test coverage that shows my fix or feature works as expected.
- [ ] I've updated or added documentation if required.
- [ ] I've included usage information in the description if CLI behavior was updated or added.
- [ ] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

File: integration/flags/.snapshots/TestInitCommand-init [hunk 1]
```diff
@@ -24,6 +24,20 @@ scan:
...
             metavars: {}
             stored: false
+        detect_rails_cookies:
+            disabled: false
+            type: risk
+            languages:
+                - ruby
+            patterns:
+                - |
+                  cookies[...] = $ANYTHING
```
Comment (cfabianski) [inline integration/flags/.snapshots/TestInitCommand-init]: Does it work if you do `cookies.signed[...] = $ANYTHING`, `cookies.permanent[...] = $ANYTHING` and so on?

Comment (elsapet) [reply integration/flags/.snapshots/TestInitCommand-init]: I'll double check but I think it does, yes. In which case, I'll add them hardcoded for now? 
Do we want all permutations of the combinations e.g. `cookies.signed.permanent` / `cookies.permanent.signed`, etc?

Comment (cfabianski) [reply integration/flags/.snapshots/TestInitCommand-init]: Yes let's do that for now

File: integration/custom_detectors/testdata/ruby/detect_rails_cookies.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+cookies.signed[:user_email] = "mish@bearer.sh"
+cookies.encrypted[:full_name] = "John Doe"
+cookies.permanent[:first_name] = "John"
+cookies[:lat_lon] = JSON.generate([47.68, -122.37])
+
+# Not detected (yet - variable reconciliation)
+# user = {
+#   first_name: "John",
+# 	last_name: "Doe"
+# }
```
Comment (cfabianski) [inline integration/custom_detectors/testdata/ruby/detect_rails_cookies.rb]: Variable reconciliation should now be supported 🙏

Comment (elsapet) [reply integration/custom_detectors/testdata/ruby/detect_rails_cookies.rb]: Looks good ✅ Scanning the following file: 

```
user = {
  first_name: "mish",
	gender_identity: "bear"
}

cookies[:login] = { value: user.to_json, expires: 1.hour, secure: true }

```

gives the following policy breaches: 

```
CRITICAL: Cookie leaking policy breach with Personal data
Cookie leaks detected

File: ../../Desktop/example/variable_reconciliation.rb:2

 6 cookies[:login] = { value: user.to_json, expires: 1.hour, secure: true }


HIGH: Cookie leaking policy breach with Sensitive data
Cookie leaks detected

File: ../../Desktop/example/variable_reconciliation.rb:3

 6 cookies[:login] = { value: user.to_json, expires: 1.hour, secure: true }

=====================================

Policy breaches detected

9 policies were run and 2 breaches were detected.

CRITICAL: 1 (Cookie leaking)
HIGH: 1 (Cookie leaking)
MEDIUM: 0
LOW: 0
```

**Review by cfabianski — APPROVED** [review]

**Review by cfabianski — APPROVED** [review]

**Review by cfabianski — APPROVED** [review]

**Pull request merged**